File: password_generator.py
"""
============================================================
 PASSWORD GENERATOR - Python (CLI + Flask Backend)
============================================================
 Lines of Code: ~120
 Run CLI : python password_generator.py
 Run API : python password_generator.py server
============================================================
"""
import random
import string
import sys
import secrets
from flask import Flask, jsonify, request

# ---------- Backend Logic ----------
def generate_password(length=12, use_upper=True, use_digit=True,
                      use_symbol=True) -> dict:
    pool = string.ascii_lowercase
    if use_upper:  pool += string.ascii_uppercase
    if use_digit:  pool += string.digits
    if use_symbol: pool += string.punctuation

    # Guarantee at least one of each requested class
    pwd = []
    if use_upper:  pwd.append(secrets.choice(string.ascii_uppercase))
    if use_digit:  pwd.append(secrets.choice(string.digits))
    if use_symbol: pwd.append(secrets.choice(string.punctuation))

    pwd += [secrets.choice(pool) for _ in range(length - len(pwd))]
    random.shuffle(pwd)
    password = ''.join(pwd)
    return {"password": password, "strength": _strength(password)}

def _strength(p: str) -> str:
    score = sum([
        len(p) >= 8,
        len(p) >= 12,
        any(c.isupper() for c in p),
        any(c.isdigit() for c in p),
        any(c in string.punctuation for c in p),
    ])
    return {0: "VERY WEAK", 1: "VERY WEAK",
            2: "MODERATE",   3: "MODERATE",
            4: "STRONG",     5: "VERY STRONG"}[score]

# ---------- CLI Mode ----------
def cli():
    print("\n========== PASSWORD GENERATOR (Python) ==========")
    length = int(input("Length (4-64, default 12): ") or 12)
    u = input("UPPERCASE? (y/n): ").lower() == 'y'
    d = input("Digits?    (y/n): ").lower() == 'y'
    s = input("Symbols?   (y/n): ").lower() == 'y'
    res = generate_password(length, u, d, s)
    print(f"\nPassword : {res['password']}")
    print(f"Strength : {res['strength']}")
    print("==================================================\n")

# ---------- Flask Backend API ----------
app = Flask(__name__)

@app.route('/api/generate', methods=['POST'])
def api_generate():
    data = request.get_json(force=True) or {}
    res = generate_password(
        length=int(data.get('length', 12)),
        use_upper=bool(data.get('upper', True)),
        use_digit=bool(data.get('digit', True)),
        use_symbol=bool(data.get('symbol', True)),
    )
    return jsonify(res)

@app.route('/')
def home():
    return '''
    <h2>Password Generator API</h2>
    POST /api/generate  JSON {"length":12,"upper":true,"digit":true,"symbol":true}
    '''

if __name__ == '__main__':
    if len(sys.argv) > 1 and sys.argv[1] == 'server':
        app.run(debug=True, port=5000)
    else:
        cli()

File: password_generator.cpp
/*
 * ============================================================
 *  PASSWORD GENERATOR - C++ (OOP)
 *  Backend/CLI Implementation
 * ============================================================
 *  Lines of Code: ~110
 *  Compile: g++ password_generator.cpp -o pgen && ./pgen
 * ============================================================
 */
#include <iostream>
#include <string>
#include <vector>
#include <random>
#include <algorithm>
#include <cctype>

class PasswordGenerator {
private:
    std::string lower = "abcdefghijklmnopqrstuvwxyz";
    std::string upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    std::string digit = "0123456789";
    std::string symbol = "!@#$%^&*()-_=+[]{};:,.<>/?";
    std::string pool;

    std::random_device rd;
    std::mt19937 gen;

public:
    PasswordGenerator() : gen(rd()) {}

    std::string generate(int length, bool useUpper,
                         bool useDigit, bool useSymbol) {
        pool = lower;
        if (useUpper)  pool += upper;
        if (useDigit)  pool += digit;
        if (useSymbol) pool += symbol;

        std::uniform_int_distribution<> dis(0, (int)pool.size() - 1);
        std::string pwd;
        pwd.reserve(length);

        // Ensure at least one of each requested type
        if (useUpper)  pwd += upper[dis(gen)];
        if (useDigit)  pwd += digit[dis(gen)];
        if (useSymbol) pwd += symbol[dis(gen)];

        while ((int)pwd.size() < length)
            pwd += pool[dis(gen)];

        std::shuffle(pwd.begin(), pwd.end(), gen);
        return pwd;
    }

    std::string strength(const std::string& pwd) {
        int score = 0;
        if (pwd.size() >= 8)  score++;
        if (pwd.size() >= 12) score++;
        if (std::any_of(pwd.begin(), pwd.end(),
                        [](char c){ return std::isupper(c); })) score++;
        if (std::any_of(pwd.begin(), pwd.end(),
                        [](char c){ return std::isdigit(c); })) score++;
        if (std::any_of(pwd.begin(), pwd.end(),
                        [](char c){ return std::ispunct(c); })) score++;
        switch (score) {
            case 0: case 1: return "VERY WEAK";
            case 2: case 3: return "MODERATE";
            case 4:        return "STRONG";
            default:        return "VERY STRONG";
        }
    }
};

int main() {
    int length; char u, d, s;
    PasswordGenerator gen;

    std::cout << "\n========== PASSWORD GENERATOR (C++) ==========\n";
    std::cout << "Length (4-64): ";       std::cin >> length;
    std::cout << "UPPERCASE? (y/n): ";    std::cin >> u;
    std::cout << "Digits?    (y/n): ";    std::cin >> d;
    std::cout << "Symbols?   (y/n): ";    std::cin >> s;

    std::string pwd = gen.generate(length, u=='y', d=='y', s=='y');
    std::cout << "\nPassword : " << pwd  << "\n";
    std::cout << "Strength: " << gen.strength(pwd) << "\n";
    std::cout << "============================================\n";
    return 0;
}
